tests/unit/browser/test_mhtml.py

```diff
@@ -0,0 +1,289 @@
+
+        file=20t
+        -----=_qute-UUID
+        Content-Location: http://z.example.com/
+        MIME-Version: 1.0
+        Content-Type: text/plain
+        Content-Transfer-Encoding: quoted-printable
+
+        file=20z
+        -----=_qute-UUID--
+        """)
+
+
+def test_empty_content_type(checker):
+    writer = mhtml.MHTMLWriter(root_content=b'',
+                               content_location='http://example.com/',
+                               content_type='text/plain')
+    writer.add_file('http://example.com/file', b'file content')
+    writer.write_to(checker.fp)
+    checker.expect("""
+        Content-Type: multipart/related; boundary="---=_qute-UUID"
+        MIME-Version: 1.0
+
+        -----=_qute-UUID
+        Content-Location: http://example.com/
+        MIME-Version: 1.0
+        Content-Type: text/plain
+        Content-Transfer-Encoding: quoted-printable
+
+
+        -----=_qute-UUID
+        MIME-Version: 1.0
+        Content-Location: http://example.com/file
+        Content-Transfer-Encoding: quoted-printable
+
+        file=20content
+        -----=_qute-UUID--
+        """)
+
+
+def test_removing_file_from_mhtml(checker):
+    writer = mhtml.MHTMLWriter(root_content=b'root',
+                               content_location='http://example.com/',
+                               content_type='text/plain')
+    writer.add_file('http://evil.com/', b'file content')
+    writer.remove_file('http://evil.com/')
+    writer.write_to(checker.fp)
+    checker.expect("""
+        Content-Type: multipart/related; boundary="---=_qute-UUID"
+        MIME-Version: 1.0
+
+        -----=_qute-UUID
+        Content-Location: http://example.com/
+        MIME-Version: 1.0
+        Content-Type: text/plain
+        Content-Transfer-Encoding: quoted-printable
+
+        root
+        -----=_qute-UUID--
+        """)
+
+
+@pytest.mark.parametrize('style, expected_urls', [
+    ("@import 'default.css'", ['default.css']),
+    ('@import "default.css"', ['default.css']),
+    ("@import \t 'tabbed.css'", ['tabbed.css']),
+    ("@import url('default.css')", ['default.css']),
+    ("""body {
+    background: url("/bg-img.png")
+    }""", ['/bg-img.png']),
+    ('background: url(folder/file.png)', ['folder/file.png']),
+    ('content: url()', []),
+])
+def test_css_url_scanner(style, expected_urls):
+    expected_urls.sort()
+    urls = mhtml._get_css_imports(style)
+    urls.sort()
+    assert urls == expected_urls
+
+
+def test_noclose_bytesio_fake_close():
+    fp = mhtml._NoCloseBytesIO()
+    fp.write(b'Value')
+    fp.close()
+    assert fp.getvalue() == b'Value'
+    fp.write(b'Eulav')
+    assert fp.getvalue() == b'ValueEulav'
+
+
+def test_noclose_bytesio_actual_close():
+    fp = mhtml._NoCloseBytesIO()
+    fp.write(b'Value')
+    fp.actual_close()
+    with pytest.raises(ValueError) as excinfo:
+        fp.getvalue()
+        assert 'I/O operation on closed file.' == str(excinfo.value)
```

Comment:
Can you dedent those asserts to be outside of the `with`-block? I'm not sure if `pytest.raises` does something important in `__exit__`.

Also, why the [Yoda conditions](https://en.wikipedia.org/wiki/Yoda_conditions)? :wink:

![yoda](https://cloud.githubusercontent.com/assets/625793/10093221/e7d71686-634f-11e5-9b9c-8bf779efc18d.png) (SCNR)